### PR TITLE
feat: [DEL-5113]: add custom role support

### PR DESCRIPTION
**Pull request by AnupamIO**
Changes:
1- User can add customRole in kubernetesPermissionType, condition is they should create the role first in required namespace then add role to k8PermissionType field

Testing:
1- Bought up a delegate with namespace_admin permission, this will create role. harness-delegate-ng-namespace-admin in mentioned namespace.
2- Install another delegate in same namespace now specify harness-delegate-ng-namespace-admin pre-created role in k8PermissionType. 
3- Execute task on delegate and checked.

### CODE REVIEW

File: harness-delegate-ng/values.yaml [hunk 1]
```diff
@@ -68,6 +68,7 @@ description: ""
 tags: ""
 
 # Permissions for installed delegate, could be CLUSTER_ADMIN, CLUSTER_VIEWER or NAMESPACE_ADMIN
+# For using custom role: Create role in kubernetes cluster and refer role in k8sPermissionsType field.
```
Comment (xingchi-jin) [inline harness-delegate-ng/values.yaml:71]: provide an example in the comment here, how to refer a role

File: harness-delegate-ng/templates/_helpers.tpl [hunk 2]
```diff
@@ -68,6 +68,18 @@ Fetch access level using kubernetes permission value
   {{- end }}
 {{- end }}
 
+
+{{/*
+Check if custom role is provided in k8sPermissionsType
+*/}}
+{{- define "harness-delegate-ng.useCustomRole" -}}
```
Comment (xingchi-jin) [inline harness-delegate-ng/templates/_helpers.tpl:75]: if user manually set useCustomRole in value.yaml file, which one takes precedence ? 
a buggy setting could be 
useCustomRole: true
k8sPermissionsType: CLUSTER_ADMIN

**Review by xingchi-jin — CHANGES_REQUESTED** [review]

File: harness-delegate-ng/templates/_helpers.tpl (hunk 2 above)
Comment (AnupamIO) [reply harness-delegate-ng/templates/_helpers.tpl:75]: There's is no setting for useCustomRole: true. If something else is mentioned apart from CLUSTER_ADMIN, CLUSTER_VIEWER or NAMESPACE_ADMIN. We will treat it like its a custom role.

**Comment by bharatgoelharness** [comment]
@AnupamIO What all custom role we are going to support?

**Comment by AnupamIO** [comment]
> @AnupamIO What all custom role we are going to support?

We don't have min set of exact permissions req because that would depend on tasks that customer wants to execute.

**Review by raghuAtWings — APPROVED** [review]

**Pull request merged**